### PR TITLE
Change Title's inner value and message type to `Text`

### PR DESCRIPTION
**Pull request by Iaiao**
# Change Title's inner value and message type to `Text`

## Status

- [x] Ready 
- [ ] Development
- [ ] Hold

## Description

This PR changes `Title` to use `Text` instead of json-serialized `String` and messages to use `Text` instead of raw `String`, adds a new quill->host serialization method (json). This change was made because bincode ^1 doesn't support `deserialize_any` but `Text` is an untagged enum and requires `deserialize_any`.

## Checklist

- [x] Ran `cargo fmt`, `cargo clippy --all-targets`, `cargo build --release` and `cargo test` and fixed any generated errors!
- [x] Removed unnecessary commented out code
- [x] Used specific traces (if you trace actions please specify the cause i.e. the player)

Note: if you locally don't get any errors, but GitHub Actions fails (especially at `clippy`) you might want to check your rust toolchain version. You can then feel free to fix these warnings/errors in your PR.

### CODE REVIEW

File: feather/plugin-host/src/host_calls/entity.rs [hunk 1]
```diff
@@ -17,12 +17,11 @@ pub fn entity_send_message(
     message_ptr: PluginPtr<u8>,
     message_len: u32,
 ) -> anyhow::Result<()> {
-    let message = cx.read_string(message_ptr, message_len)?;
+    let message = cx.read_json(message_ptr, message_len)?;
```
Comment (Defman) [inline feather/plugin-host/src/host_calls/entity.rs:20]: We are blindly trusting plugins to send valid text messages as json without confirming?

Comment (Defman) [reply feather/plugin-host/src/host_calls/entity.rs:20]: I think this is fine for now, but shows that the current `Text` api is a bit lacking...

Comment (Iaiao) [reply feather/plugin-host/src/host_calls/entity.rs:20]: The text component is serialized with serde_json in quill/api

File: quill/api/src/entity.rs [hunk 2]
```diff
@@ -92,16 +93,16 @@ impl Entity {
     ///
     /// The message sends as a "system" message.
     /// See [the wiki](https://wiki.vg/Chat) for more details.
-    pub fn send_message(&self, message: impl AsRef<str>) {
-        let message = message.as_ref();
+    pub fn send_message(&self, message: impl Into<Text>) {
+        let message = message.into().to_string();
```
Comment (Defman) [inline quill/api/src/entity.rs:97]: We should define some lint which either prefers `to_string()` or `to_owned()`.

**Review by Defman — APPROVED** [review]

File: quill/api/src/entity.rs (hunk 2 above)
Comment (Defman) [reply quill/api/src/entity.rs:97]: It seems that clippy already lints `"some_string_literal".to_string()` and suggests `to_owned()`.

Comment (Iaiao) [reply quill/api/src/entity.rs:97]: `.into()` returns `Text` and not `&str`. `to_string()` is derived from `Text`'s `Display` trait implementation, which serializes the text component with `serde_json` and returns the json string.

**Pull request merged**